draft-steele-cose-merkle-tree-proofs.md

```diff
@@ -60,188 +60,73 @@ informative:
 
-Merkle proofs can be used to prove a document is in a database (proof of inclusion),
-or that a smaller set of statements are contained in a large set of statements (selective disclosure proofs).
+Verifiable data structure proofs can be used to prove a document is in a database (proof of inclusion),
+that a database is append only (proof of consistency), that a smaller set of statements are contained
+in a large set of statements (proof of disclosure, a special case of proof of inclusion),
+or proof that certain data is not yet present in a database (proofs of non inclusion).
 
-Typically, merkle trees are constructed from simple operations such as concatenation and digest via a cryptographic hash function.
+Differences in the representation of verifiable data structures, and verifiable data structure proof types,
+can increase the burden for implementers, and create interoperability challenges for transparency services.
 
-The simple design and valuable cryptographic properties of merkle trees have been leveraged in many network and database applications.
-
-Differences in the representation of a merkle tree, merkle leaf and merkle inclusion proof can increase the
-burden for implementers, and create interoperability challenges.
-
-This document describes the three data structures necessary to use merkle proofs with COSE envelopes.
+This document describes how to convey verifiable data structures, and associated proof types in COSE envelopes.
 
 ## Requirements Notation
 
 {::boilerplate bcp14-tagged}
 
 # Terminology
 
-Leaf:
-
-: A merkle tree leaf is the cryptographic hash of a sequence of bytes that combines Leaf Payload and Extra Data.
-
-Merkle Tree:
-
-: A Merkle tree is a tree where every leaf is a cryptographic hash of a sequence of
-bytes and every node that is not a leaf is the cryptographic hash of the its child nodes.
-
-Merkle Root:
+Verifiable Data Structure:
 
-: A Merkle root is the root node of a tree; this cryptographic hash is a committment to the content of the tree.
+: A data structure which supports one or more Proof Types.
 
-Merkle Tree Algorithm:
+Proof Type:
 
-: A Merkle tree algorithm specifies how to construct the tree and how to compute its root.
+: A verifiable process, that proves properties of one or more Verifiable Data Structures.
 
-Leaf Payload and Extra Data:
+Proof Value:
 
-: A leaf payload is application data used to produce a Leaf.
-The Merkle tree algorithm determines how a payload together with extra data is used to produce a leaf.
-The simplest case is that the leaf is the cryptographic hash of the payload without extra data.
+: An encoding of a Proof Type in CBOR.
 
-Inclusion Path:
+Proof Signature:
 
-: An inclusion path enables a verifier to recompute a root, given a leaf and extra data.
+: A COSE Sign1 encoding of a specific Proof Type for a specific Verifiable Data Structure.
 
-Inclusion Proof:
+# Verifiable Data Structures in CBOR {#sec-generic-verifiable-data-structures}
```

Comment:
This is the beginning of changes after #20 

Substantial reordering an rewording follows...

Reply:
I've separately approved #19 and #20 in other PRs.Starting review from this point